core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -689,6 +689,7 @@ object LogSegment {
   }
 }
 
-object LogFlushStats extends KafkaMetricsGroup {
-  val logFlushTimer: Timer = newTimer("LogFlushRateAndTimeMs", TimeUnit.MILLISECONDS, TimeUnit.SECONDS)
+object LogFlushStats {
+  private val metricsGroup = new KafkaMetricsGroup(this.getClass)
```

Comment:
Let's pass `classOf[LogFlushStats]`.

Reply:
Since it's an object without the class, `classOf` won't work. I replaced with `LogFlushStats.getClass` here and in other similar places you mentioned.